File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution 
{
    public:
        string longestCommonPrefix(vector<string>& strs) 
        {
            if(strs.empty())
				return "";

			for(int i = 0; i < static_cast<int>(strs[0].length()); ++i)
			{
				const auto c = strs[0][i];

				for(int j = 1; j < static_cast<int>(strs.size()); ++j)
				{
					if(i >= static_cast<int>(strs[j].length()) || strs[j][i] != c)
						return strs[0].substr(0, i);
				}
			}

			return strs[0];
        }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.cs
public class Solution 
{
    public string LongestCommonPrefix(string[] strs) 
    {
        if (strs.Length == 0) 
            return "";
        
        for (var i = 0; i < strs[0].Length; i++) 
        {
            var c = strs[0][i];

            for (var j = 1; j < strs.Length; j++) 
            {
                if (i >= strs[j].Length || strs[j][i] != c)
                    return strs[0][..i];
            }
        }
        
        return strs[0];        
    }
}